.github/workflows/test.yml

```diff
@@ -0,0 +1,32 @@
+name: Test
+on:
+  - pull_request
+  - push
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ['3.8', '3.9', '3.10']
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
```

Comment:
we need to cache the dependencies for a faster run. Can you do some research and implement this caching mechanism, please?
Here is an example : [caching dep example](https://github.com/sudiptob2/cf-stats/blob/befabb28dfb41e34a5151fad1898842152c3cb33/.github/workflows/test.yml#L23)